File: lecture_012(Sorting)/l_013_sort01.java
import java.util.Scanner;
public class l_013_sort01 {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
		int[] arr = new int[n];
		input(arr);
        display(partitionSort01(arr));
    }
    public static void display(int[] arr) {
        for(int r: arr) {
            System.out.print(r+" ");
        }
        System.out.println();
    }
    public static void input(int[] arr) {
        for(int i=0; i<arr.length; i++) {
            arr[i] = scn.nextInt();
        }
        display(arr);
    }
    public static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
    public static int[] partitionSort01(int[] arr) {
        int i=0;
        int j=0;
        while(i<arr.length) {
            if(arr[i]==1) {
                i++;
            }
            else {
                swap(arr,i,j);
                i++;
                j++;
            }
        }
        return arr;
    }
}

File: lecture_012(Sorting)/l_013_sort01.cpp
#include<iostream>
#include<vector>
using namespace std;

void display(vector<int> &arr) {
    for(int r: arr) {
        cout<<r<<" ";
    }
    cout<<endl;
}
void input(vector<int> &arr) {
    for(int i=0; i<arr.size(); i++) {
        cin>>arr[i];
    }
    display(arr);
}
void swap(vector<int> &arr, int i ,int j) {
	int temp = arr[i];
	arr[i] = arr[j];
	arr[j] = temp;
}
vector<int> partitionsort01(vector<int> &arr) {
	int i=0;
	int j=0;
	while(i<arr.size()) {
		if(arr[i]==1) {
			i++;
		}
		else {
			swap(arr,i,j);
			i++;
			j++;
		}
	}
	return arr;
}
int main() {
    int n; cin>>n;
    vector<int> arr(n,0);
    input(arr);
    vector<int> result = partitionsort01(arr);
    display(result);

	return 0;
}